src/BSON.jl

```diff
@@ -14,16 +14,16 @@ const Primitive = Union{Nothing,Bool,Int32,Int64,Float64,String,Vector{UInt8},BS
   datetime, null, regex, dbpointer, javascript, symbol, javascript_scoped,
   int32, timestamp, int64, decimal128, minkey=0xFF, maxkey=0x7F)
 
-applychildren!(f, x) = x
+applychildren!(::Function, x) = x
```

Comment:
How about `Base.Callable` ?

Reply:
Oh OK, I thought it was a optimisation because `Callable` is a concrete type AFAICT, but `Function` is abstract.

I think `applychildren!` is only used internally. It is not meant as an external interface, so it deosn't need to be more general. In fact, I would prefer it to be more specific, but I am not sure if Function accepts type parameters for specifying the function arguments and return value.